apis/apiextensions/v1beta1/composition_types.go

```diff
@@ -102,7 +112,70 @@ type ComposedTemplate struct {
 	ReadinessChecks []ReadinessCheck `json:"readinessChecks,omitempty"`
 }
 
-// TypeReadinessCheck is used for readiness check types
+// mapPatchSets processes a list of PatchSets and returns a map of PatchSet
+// ID to Patches.
+func (c *ComposedTemplate) mapPatchSets(patchSets []PatchSet) (patchSetMapType, error) {
```

Comment:
Nope. Split this out into a separate function to avoid cyclomatic complexity on `InjectCompositionData()` but it doesn't need access to `c` so can be a standalone private function. Will fix.